lib/resources/zfs_dataset.rb

```diff
@@ -0,0 +1,62 @@
+# encoding: utf-8
+# author: Joseph Benden
+
+module Inspec::Resources
+  class ZfsDataset < Inspec.resource(1)
+    name 'zfs_dataset'
+    desc "
+      Use the zfs_dataset InSpec audit resource to test if the named
+      ZFS Dataset is present and/or has certain properties.
+    "
+    example "
+      describe zfs_dataset('tank/tmp') do
+        its('exec') { should eq('off') }
+        its('setuid') { should eq('off') }
+      end
+    "
+
+    def initialize(zfs_dataset)
+      return skip_resource 'The `zfs_dataset` resource is not supported on your OS yet.' if !inspec.os.bsd?
```

Comment:
We don't need to account for it in this PR, but I suspect this resource would work OK as-is on Solaris and any Linux that is using ZFS-on-Linux too, with the small exception of the actual ZFS binary paths may be different. This is worth some investigation.